### DISCUSSION TITLE
Help requested regarding Socket.IO usage with protocol: Domoticz?

### DISCUSSION THREAD

**Discussion by rgroothuis** [socketio/socket.io#4715] [Q&A] [answered]
In a react native application I've the following line which is working fine: 

`var ws = new WebSocket(<<URL>> + '/json', ["domoticz"])`

What is relevant is the second parameter: ["domoticz"]. Without this parameter I cannot make a successful connection to the Domoticz server. 

Now I want to use the Socket.IO client because it has the reestablishment option when the connection drops. 

Question: How to pass the ["domoticz"] paramater in the Socket.IO connection call? 

If tried many combinations but I cannot get it to work. One of the variations is: 

   ```
 function onConnect() {
      console.log('Connected!!!!')
    }
    function onDisconnect() {
      console.log('Disconnected!!!!')
    }
    const socket = io(domoticzData.address + ':' + domoticzData.port + '/json',
      {protocols: ["domoticz"]}
    );
    socket.on('connect', onConnect);
    socket.on('disconnect', onDisconnect);

    console.log('Socket: ', socket)
```

I don't see the connect message in my console!

What am I doing wrong?

**Comment by darrachequesne** [comment, marked as answer]
Please note that the Socket.IO client is not a WebSocket client, so unless your server is a Socket.IO server, it won't work.

You might be interested in https://github.com/nathanboktae/robust-websocket

See also: https://socket.io/docs/v4/#what-socketio-is-not
